tools/dev-server/index.bzl

```diff
@@ -0,0 +1,125 @@
+load("@build_bazel_rules_nodejs//internal/common:sources_aspect.bzl", "sources_aspect")
+
+"""
+  Gets the workspace name of the given rule context.
+"""
+
+def _get_workspace_name(ctx):
+    if ctx.label.workspace_root:
+        # We need the workspace_name for the target being visited.
+        # Skylark doesn't have this - instead they have a workspace_root
+        # which looks like "external/repo_name" - so grab the second path segment.
+        return ctx.label.workspace_root.split("/")[1]
+    else:
+        return ctx.workspace_name
+
+"""
+  Implementation of the dev server rule.
+"""
+
+def _dev_server_rule_impl(ctx):
+    files = depset(ctx.files.srcs)
+    required_tools = ctx.files._bash_runfile_helpers + \
```

Comment:
Are the `\`s required?  Since `bzl` is based on python syntax I don't think it would be necessary?

Reply:
Yeah, those are required in order to be able to wrap the list concatenation. At least, bazel complained if I just wrapped them 🤷‍♂